src/libs/SidebarUtils.ts

```diff
@@ -457,7 +443,108 @@ function getOptionData({
     return result;
 }
 
+function getWelcomeMessage(report: OnyxEntry<Report>): WelcomeMessage {
+    const welcomeMessage: WelcomeMessage = {showReportName: true};
+    if (ReportUtils.isChatRoom(report)) {
+        return getRoomWelcomeMessage(report);
+    }
+
+    if (ReportUtils.isPolicyExpenseChat(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartOne');
+        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartTwo');
+        welcomeMessage.phrase3 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartThree');
+        if (report?.description) {
+            welcomeMessage.concatMessage = parseHtmlToText(report.description);
+        }
+        welcomeMessage.concatMessage = `${welcomeMessage.phrase1} ${ReportUtils.getDisplayNameForParticipant(report?.ownerAccountID)} ${welcomeMessage.phrase2} ${ReportUtils.getPolicyName(
+            report,
+        )} ${welcomeMessage.phrase3}`;
+        return welcomeMessage;
+    }
+
+    if (ReportUtils.isSelfDM(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistorySelfDM');
+        welcomeMessage.concatMessage = welcomeMessage.phrase1;
+        return welcomeMessage;
+    }
+
+    if (ReportUtils.isSystemChat(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistorySystemDM');
+        welcomeMessage.concatMessage = welcomeMessage.phrase1;
+        return welcomeMessage;
+    }
+
+    if (
+        !(ReportUtils.isChatRoom(report) || ReportUtils.isPolicyExpenseChat(report) || ReportUtils.isSelfDM(report) || ReportUtils.isInvoiceRoom(report) || ReportUtils.isSystemChat(report))
+    ) {
```

Comment:
Checking this again we should remove the condition as it's useless. All the conditions inside are used above (isInvoiceRoom is a part of isChatRoom)